students/Zach Cooper/lesson10/database.py

```diff
@@ -0,0 +1,198 @@
+"""
+    Create Database Using MongoDB
+"""
+import csv
+from pymongo import MongoClient
+import time
+import csv
+
+
+class MongoDBConnection():
+    """ Set up MongoDB Connection and then close it"""
+
+    def __init__(self, host='127.0.0.1', port=27017):
+        """ Windows needs to connect to host IP Address """
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_tb):
+        """ Closes Database Connecton"""
+        self.connection.close()
+
+
+def timed_function(func):
+    """
+    This function times stuff
+    """
+    def decorate_func(*args, **kwargs)
+        start_time = time.time()
+        result = func(*args, **kwargs)
+        end_time = time.time()
+        total_time = end_time - start_time
+        if total_time > 0:
+            time_str = f'{func.__nam__},{total_time}\n'
+            with open("timing.csv", "a") as file:
+                file.write(time_str)
+        return result
+
+
+class Time_stuff(type):
+    """
+        Example used from website we viewed online
+    """
+    def__new__(cls, name, bases, attr):
+        for name, value in attr.items():
+            if type(value) is types.FunctionType or type(
+                    value) is types.MethodType:
+                attr[name] = time_func(value)
+            return super(Time_stuff, cls).__new__(cls, name, bases, attr)
```

Comment:
Looks pretty good, but not using it anywhere. To make this work, you need to refactor your basic functions to be in a class, and then you could use the metaclass.